controllers/skills.js

```diff
@@ -0,0 +1,105 @@
+const skills = require("../models/skills");
+const logger = require("../utils/logger");
+
+/**
+ * Award a skill
+ *
+ * @param req {Object} - Express request object
+ * @param req.body {Object} - User object
+ * @param res {Object} - Express response object
+ */
+async function awardSkill(req, res) {
+  let addedData;
+
+  try {
+    const { username } = req.params;
+    addedData = await skills.awardSkill(req.body, username);
+  } catch (error) {
+    logger.error("Error posting skill data: ", error);
+    return res.boom.notFound("User doesn't exist");
+  }
+
+  return res.json({
+    message: "Added data successfully!",
+    content: addedData,
+  });
+}
+
+/**
+ * Fetch all skills
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+async function fetchSkills(req, res) {
+  let allSkills;
+
+  try {
+    allSkills = await skills.fetchSkills();
+  } catch (error) {
+    logger.error("Error fetching skill data: ", error);
+    return res.boom.serverUnavailable("Something went wrong please contact admin");
+  }
+
+  return res.json({
+    message: "Skills returned successfully!",
+    skills: allSkills.length > 0 ? allSkills : [],
+  });
+}
+
+/**
+ * Fetch skills of User
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+async function fetchUserSkills(req, res) {
+  const { username } = req.params;
+
+  try {
+    const userData = await skills.fetchUserSkills(username);
+    return res.json({
+      message: "Skills returned successfully!",
+      data: userData,
+    });
+  } catch (error) {
+    logger.error("Error fetching user skills: ", error);
+    return res.boom.notFound("User doesn't exist");
+  }
+}
+
+/**
+ * Fetch users with given skill
+ *
+ * @param req {Object} - Express request object
+ * @param res {Object} - Express response object
+ */
+async function userWithSkill(req, res) {
+  let { skill } = req.params;
+  // In case user makes query with - or _ for multiple words
+  const regex = /[-_]/g;
```

Comment:
Can we please import this from a constant file ?

Reply:
just the regex? why?
I mean it's only a single line of code for a specific case...